src/sentry/api/endpoints/project_stacktrace_links.py

```diff
@@ -0,0 +1,148 @@
+from sentry.services.hybrid_cloud.integration import integration_service
+from sentry.shared_integrations.exceptions import ApiError
+
+from .project_stacktrace_link import get_code_mapping_configs
+
+MAX_CODE_MAPPINGS_USED = 3
+
+
+class StacktraceLinksSerializer(serializers.Serializer):  # type: ignore
+    file = serializers.ListField(child=serializers.CharField())
+
+    # falls back to the default branch
+    commit_id = serializers.CharField(required=False)
+
+
+@region_silo_endpoint
+class ProjectStacktraceLinksEndpoint(ProjectEndpoint):  # type: ignore
+    """
+    Returns valid links for source code providers so that
+    users can go from the file in the stack trace to the
+    provider of their choice.
+
+    Simular to `ProjectStacktraceLinkEndpoint` but allows
+    for bulk resolution.
+
+    `file`: The file path from the stack trace
+    `commitId` (optional): The commit_id for the last commit of the
+                           release associated to the stack trace's event
+    """
+
+    def get(self, request: Request, project: Project) -> Response:
+        serializer = StacktraceLinksSerializer(data=request.GET)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        data = serializer.validated_data
+
+        result = {"files": [{"file": file} for file in data["file"]]}
+
+        mappings_used = 0
+
+        configs = get_code_mapping_configs(project)
+
+        default_error = "stack_root_mismatch" if configs else "no_code_mappings"
+
+        for config in configs:
+            # find all the files that match the current code mapping's stack_root
+            # and have not already been resolved by another code mapping
+            #
+            # if the's an error from a previous code mapping attempted, but this
+            # current code mapping can be used, we should try again
+            files = [
+                file
+                for file in result["files"]
+                if file.get("sourceUrl") is None and file["file"].startswith(config.stack_root)
+            ]
+            if not files:
+                continue
+
+            # safety to limit the maximum number of mappings used
+            # to avoid reaching API rate limits
+            if mappings_used >= MAX_CODE_MAPPINGS_USED:
+                for file in files:
+                    if not file.get("error") and file.get("sourceUrl") is None:
+                        file["error"] = "max_code_mappings_applied"
+                continue
+
+            mappings_used += 1
+
+            install = get_installation(config)
+
+            # should always be overwritten
+            error: str | None = "file_not_checked"
+
+            # since the same code mapping stack root matches all these files, we only check the
+            # first file and we will assume the other matching files will resolve the same way
+            version = data.get("commit_id")
```

Comment:
```suggestion
            version = data.get("commit_id", config.default_branch)
```
Would this help the code be cleaner?

Reply:
No, so this follows a similar logic as the original stacktrace link where it tries the provided commit first, and if it doesn't exist, it falls back to using the default branch on the second attempt.

By not providing the default branch to the `.get` here, it means that we only ever check against the default branch in the second block which I think is cleaner.